mono/mini/mini-trampolines.c

```diff
@@ -822,16 +813,21 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 gpointer
 mono_magic_trampoline (mgreg_t *regs, guint8 *code, gpointer arg, guint8* tramp)
 {
-	MonoError error;
 	gpointer res;
 
+	MONO_PREPARE_RESET_BLOCKING_UNBALANCED;
+
+	MonoError error;
+
 	trampoline_calls ++;
 
 	res = common_call_trampoline (regs, code, (MonoMethod *)arg, NULL, NULL, &error);
-	if (!mono_error_ok (&error)) {
-		mono_error_set_pending_exception (&error);
-		return NULL;
-	}
+	mono_error_raise_exception (&error);
```

Comment:
I'm pretty ambivalent about `raise` vs `set_pending` in this case: mostly we need to avoid raising in functions that are deep in the call tree from an interrupt check so that we can free resources (eventually local handles, for example).  In this case the trampoline will start unwinding either way, so there's no hidden control flow.

As a note, you can do:
```
if (mono_error_set_pending_exception (&error))
  goto done;
```
which is slightly shorter.